Framework/Algorithms/test/NormaliseToMonitorTest.h

```diff
@@ -441,6 +445,172 @@ class NormaliseToMonitorTest : public CxxTest::TestSuite {
+    for (size_t i = 0; i < specOutInfo.size(); ++i) {
+      const auto &yValues = outWS->histogram(i).y();
+      for (size_t j = 0; j < yValues.size(); ++j) {
+        if (specOutInfo.isMonitor(i))
+          TS_ASSERT_DELTA(yValues[j], 3.0, 1e-12)
+        else
+          TS_ASSERT_DELTA(yValues[j], 6.0 / double(j + 1), 1e-12)
+      }
+    }
+  }
+
+  void test_with_scanning_workspace_integration_range() {
+    auto testWS = makeTestDetectorScanWorkspace();
+
+    NormaliseToMonitor alg;
+    alg.setChild(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", testWS))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("OutputWorkspace", "outputWS"))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("MonitorID", "9"))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("IntegrationRangeMin", "-1"))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("IntegrationRangeMax", "99"))
+    TS_ASSERT_THROWS_NOTHING(alg.execute())
+    TS_ASSERT(alg.isExecuted())
+
+    MatrixWorkspace_sptr outWS = alg.getProperty("OutputWorkspace");
+
+    const auto &specOutInfo = outWS->spectrumInfo();
+    for (size_t i = 0; i < specOutInfo.size(); ++i) {
+      const auto &yValues = outWS->histogram(i).y();
+      for (size_t j = 0; j < yValues.size(); ++j) {
+        if (specOutInfo.isMonitor(i))
+          TS_ASSERT_DELTA(yValues[j], double(j + 1) / 15.0, 1e-12)
+        else
+          TS_ASSERT_DELTA(yValues[j], 2.0 / 15.0, 1e-12)
+      }
+    }
+  }
+
+  void test_with_async_scan_workspace_throws() {
+    const size_t N_DET = 10;
+    const size_t N_BINS = 5;
+
+    // Set up 2 workspaces to be merged
+    auto ws1 = WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(
+        N_DET, N_BINS, true);
+    auto ws2 = WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(
+        N_DET, N_BINS, true);
+    auto &detInfo1 = ws1->mutableDetectorInfo();
+    auto &detInfo2 = ws2->mutableDetectorInfo();
+    for (size_t i = 0; i < N_DET; ++i) {
+      detInfo1.setScanInterval(i, {10, 20});
+      detInfo2.setScanInterval(i, {20, 30});
+    }
+    // Merge
+    auto merged = WorkspaceFactory::Instance().create(ws1, 2 * N_DET);
+    auto &detInfo = merged->mutableDetectorInfo();
+    detInfo.merge(detInfo2);
+    merged->setIndexInfo(Indexing::IndexInfo(merged->getNumberHistograms()));
+
+    NormaliseToMonitor alg;
+    alg.setChild(true);
+    alg.setRethrows(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", merged))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("OutputWorkspace", "outputWS"))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("MonitorID", "9"))
+    TS_ASSERT_THROWS_EQUALS(alg.execute(), std::runtime_error & e,
+                            std::string(e.what()),
+                            "More then one spectra corresponds to the "
+                            "requested monitor ID, which is unheard of")
```

Comment:
"unheard of" sounds a bit too colloquial?

Reply:
This made me laugh the first time I triggered when trying with a scanning workspace. I'll change it to something more in keeping with the normal error messages.